### PR TITLE
Only create/update cache if there are new entries

### PR DESCRIPTION
**Pull request by Jguer**
- ggshield always creates/updates cache even if there are no results. Modify this behavior so `.cache_ggshield` is only written if there are entries in the cache

- also fix tests writing to the same file in the fs. they should run in isolation now with `pyfakefs`

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -454,3 +455,12 @@ def cli_runner():
...
     with cli_runner.isolated_filesystem():
         yield cli_runner
+
+
+@pytest.fixture(scope="function")
+def isolated_fs(fs):
+    # isolate fs but include CA bundle for https validation
+    fs.add_real_directory(os.path.dirname(extract_zipped_paths(DEFAULT_CA_BUNDLE_PATH)))
+    # add cassetes dir
+    cassetes_dir = join(dirname(realpath(__file__)), "cassettes")
```
Comment (agateau-gg) [inline tests/conftest.py:465]: typo: casset**t**es

File: tests/test_config.py [hunk 2]
```diff
@@ -172,12 +172,20 @@ def test_save_cache(self, cli_fs_runner):
             file_content = json.load(file)
             assert file_content == {"last_found_secrets": ["XXX"]}
 
+    @pytest.mark.parametrize("with_entry", [True, False])
     @patch("ggshield.config.Config.CONFIG_LOCAL", [".gitguardian.yml"])
-    def test_save_cache_first_time(self, cli_fs_runner):
-        os.remove(".cache_ggshield")
+    def test_save_cache_first_time(self, isolated_fs, with_entry):
+        """
+        GIVE no existing cache
```
Comment (agateau-gg) [inline tests/test_config.py:179]: typo
```suggestion
        GIVEN no existing cache
```

File: ggshield/config.py [hunk 3]
```diff
@@ -280,6 +280,10 @@ def to_dict(self) -> Dict[str, Any]:
         return _cache
 
     def save(self) -> bool:
+        if len(self.last_found_secrets) == 0:
```
Comment (agateau-gg) [inline ggshield/config.py:283]: PEP 8 says this should be `if not self.last_found_secrets:`

**Review by agateau-gg — CHANGES_REQUESTED** [review]
Minor remarks, looks good otherwise.

File: tests/conftest.py (hunk 1 above)
Comment (Jguer) [reply tests/conftest.py:465]: ? seems to be ok here

Comment (agateau-gg) [reply tests/conftest.py:465]: I meant the variable name (and matching comment)

Comment (Jguer) [reply tests/conftest.py:465]: fair point, didn't notice it

**Review by agateau-gg — APPROVED** [review]

**Pull request merged**